dbms/src/AggregateFunctions/AggregateFunctionWindowFunnelNoNullifyVariadic.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <AggregateFunctions/AggregateFunctionNull.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+class AggregateFunctionWindowFunnelNoNullifyVariadic final : public AggregateFunctionNullBase<false, AggregateFunctionWindowFunnelNoNullifyVariadic>
+{
+public:
+    AggregateFunctionWindowFunnelNoNullifyVariadic(AggregateFunctionPtr nested_function_, const DataTypes & arguments, const Array & params)
+        : AggregateFunctionNullBase<false, AggregateFunctionWindowFunnelNoNullifyVariadic>(std::move(nested_function_), arguments, params),
+        number_of_arguments(arguments.size())
+    {
+        if (number_of_arguments == 1)
+            throw Exception("Logical error: single argument is passed to AggregateFunctionWindowFunnelNoNullifyVariadic", ErrorCodes::LOGICAL_ERROR);
+
+        if (number_of_arguments > MAX_ARGS)
+            throw Exception("Maximum number of arguments for aggregate function with Nullable types is " + toString(size_t(MAX_ARGS)),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        for (size_t i = 0; i < number_of_arguments; ++i)
+            is_nullable[i] = arguments[i]->isNullable();
+    }
+
+    void add(AggregateDataPtr place, const IColumn ** columns, size_t row_num, Arena * arena) const override
+    {
+        /// This container stores the columns we really pass to the nested function.
+        const IColumn * nested_columns[number_of_arguments];
+
+        for (size_t i = 0; i < number_of_arguments; ++i)
+        {
+            if (is_nullable[i])
+            {
+                const ColumnNullable & nullable_col = assert_cast<const ColumnNullable &>(*columns[i]);
+                nested_columns[i] = &nullable_col.getNestedColumn();
```

Comment:
How exactly it will work?
It looks like that we just took inner values of Nullable column regardless to null map.

Reply:
What is out intent?
- to assume that NULL is an event but event does not match to any pattern;
- skip NULLs as no event exists.

Reply:
If second, the implementation can be more simple...

Reply:
Do I understand correctly that we know that the column is UInt8 and NULL is interpreted as 0?

Reply:
My intent is to assume that NULL is an event.

To help explain my intent, I prepared data and a select query.

```SQL
create table funnel_test (dt DateTime, user int, event Nullable(String), action Nullable(String)) engine = Log()
insert into funnel_test values (1, 1, 'e1', NULL) (2, 1, NULL, 'a1')
```
```
                 dt   user   event   action

1970-01-01 09:00:01      1   e1      NULL  
1970-01-01 09:00:02      1   NULL    a1    
```

To find the events and actions flow of `event='e1' --> action='a1'`, let's execute below query.

```SQL
select user, windowFunnel(86400)(dt, event='e1', action='a1') from funnel_test group by user
```

But the result of query is NULL without this PullRequest.

And I found that `AggregateFunctionNull` skips any row containing a null-column-value.

So I implemented `AggregateFunctionWindowFunnelNoNullifyVariadic` and use values without null-map.

If there is any problems, let me know please.

Thank you for your comment.